tests/test_app.py

```diff
@@ -113,5 +115,93 @@ def test_POST_recommendation(self, mock_text_sentiment, mock_get):
 
         self.assertEqual(204, response.status_code)
 
+
+    @patch('services.spotify_service.requests.get')
+    @patch.object(WatsonService, 'get_watson_text_analyze')
+    def test_POST_recommendation_called_with_specific_params(self, mock_sentiment, mock_get):
+        # Set up mock functions
+        file_path = 'tests/fixtures/watson_positive.json'
+        with open(file_path) as json_file:
+            ml_data = json.load(json_file)
+
+        mock_sentiment.return_value = ml_data
+
+        file_path = 'tests/fixtures/spotify_tracks.json'
+        with open(file_path) as json_file:
+            tracks_data = json.load(json_file)
+
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.json.return_value = tracks_data
+
+        # Check that Spotify API is called with correct parameters when genre setting is 'classical'
+        data = {
+            'text': 'Super positive great happy fun awesome great wonderful times',
+            'current_mood': 0,
+            'access_token': 'totally-real-access-token',
+            'genre': 'classical'
+        }
+
+        response = self.app.post('/api/v1/recommendation',
+                                 data=json.dumps(data),
+                                 content_type='application/json')
+
+        mock_get.assert_called_with(
```

Comment:
This assert_called_with test is cool!

Reply:
I know right! Python's mocking library is really powerful once you get the hang of it.